### ISSUE TITLE
Add additional data to snapshot

### ISSUE THREAD

**Issue by davist11** [sjelfull/craft-audit#12]
It would be powerful to view all (or more) of the criteria from the element being saved.

**Comment by sjelfull** [comment]
Like all the content fields? Yeah, I've thought a lot about that, and it is something I want to add.

Wasn't sure how to pull it off, though, so open to good ideas.

I could iterate over the field layout, and support all the first-party and popular third-party field types. 

Once you hit any relationship fieldtypes (Assets, Entries, Users, Categories), do you then follow that relationship and save the target elements? Do you only follow 1 point away from the source element?

**Comment by Rias500** [comment]
You could just add the base fields, then throw an event to let people add data to it from custom fields or any other things they want to add to the data from their own plugin or module.

**Comment by davist11** [comment]
Right, you have the element, so you should be able to access the properties on it. All simple field types (text/numbers/encoded JSON) should collapse down ok. It's the properties where you have to join with another table where it would start to get complex.

**Comment by sjelfull** [comment]
Sounds doable. I will add it soon.

**Comment by sjelfull** [comment]
I've pushed some initial work on this. It will use a built-in method on elements to save the content to the snapshot:

![image](https://user-images.githubusercontent.com/10508/36165148-dd4c473c-10ee-11e8-81cf-4b48bb1733fb.png)

What do you think about that? I'm probably going to hide empty keys.

Some ideas for the future:
- Diffing of changes
- Show content for related entries (now it will just show the ids, if any)

**Comment by davist11** [comment]
This looks great. I think what you've got in there for future changes make sense, especially showing some content for related items, especially the title, would be valuable.

As @Rias500 mentioned above, it may make sense to add a hook to be able to add additional data to the snapshot. For example, if we wanted to add Variant data on a Commerce site, would you want to support all the possible custom element types in this plugin? Or just add a hook for developers to add in that data?

**Comment by sjelfull** [comment]
Added an event that lets other plugins modify and add to the snapshot in #16
